tests/test_count_character_frequency.py

```diff
@@ -0,0 +1,77 @@
+import os
+from collections import Counter
+from pathlib import Path
+from typing import Generator
+
+import pytest
+from _pytest.tmpdir import TempPathFactory
+
+from characters_counter import count_character_frequency
+
+CONTENT = (
+    "Typer is a library for building CLI applications that users will love using and developers "
+    "will love creating. Typer is a library for building CLI applications that users will love using and "
+    "developers will love creating. Based on Python type hints. "
+    "It's also a command line tool to run scripts, automatically converting them to CLI applications."
+)
+
+
+class TestCharacterFrequency:
+    @pytest.mark.parametrize(
+        "data_input, result",
+        [
+            ("Python is the best language in the world", {"t": 4, "e": 4, "h": 3}),
+            ("python", {"p": 1, "y": 1, "t": 1}),
+            ("", {}),
+        ],
+    )
+    def test_count_character_frequency_with_string_input(self, data_input: Path, result: dict) -> None:
+        assert count_character_frequency(data_input) == result
+
+    @pytest.fixture(scope="function")
```

Comment:
1. function is default scope, you don't need to tell it explicitly. But please think more about it, do you need the file to be re-created for each test, or you want to have the sample file be created once for all the tests. 
2. Typically, fixtures are moved at the top of the class / file or moved to the module called conftest.py

Reply:
Aha, okey... Well, I moved a fixture to the module conftest.py and choose "session" for scope.